### ISSUE TITLE
Prints `Needs push` while actually needs pull

### ISSUE THREAD

**Issue by askhat** [fboender/multi-git-status#1]
For repositories that need to update with upstream and don't have any changes in working copy, program show status `Needs push`, while actually it should be `Needs pull`.

For example:

![screen shot 2016-12-09 at 22 48 08](https://cloud.githubusercontent.com/assets/3759947/21062389/b7eed4c2-be61-11e6-933c-86c58ebb7079.png)

Obviously I'm not that cool to write ruby core, I just keep it for reference and contemplation.

**Comment by fboender** [comment]
Thanks for the bugreport Askhat!

Can you try the latest master branch? I've committed a change that should fix this.

With kind regards

Ferry

**Comment by askhat** [comment]
Well, fix did not exactly resolved the issue. Now if repository needs to be pulled, program shows status `ok`. I assume repos needs `fetch` before `merge-base`. Unfortunately, it could take too much time for each repo. So I'd suggest you either refuse this functionality or provide some special option which will preform `fetch`.

**Comment by askhat** [comment]
Perhaps I should elaborate:
At first, after you introduced changes, I pulled your repository, and program showed status `ok` for repos which actually needed pull. I manually performed `fetch` in those directories, then program showed `Needs pull` as expected.

**Comment by fboender** [comment]
That was the intended function of the new code I committed. It's unfortunate that there are so many different states a repository can be in, and there are even more ways of trying to figure out which exact state it's in. I also need any inspection of a repo to be fast, since I usually need to get the status of a few dozen repos.

I'm going to keep the changes I made, as I feel they are an improvement, even if they're not perfect ;-)

Thank you for your report and feedback!
